pacstall/api/config_facade.py

```diff
@@ -0,0 +1,259 @@
+
+    Parameters
+    ----------
+    repo_root_url (str): URL to the root of the repository.
+
+    Returns
+    -------
+    `True` if the repository respects the file structure, otherwise `False`.
+    """
+    try:
+        # Choose branch by checking if the `packagelist` file exists
+        with get(f"{repo_root_url}/packagelist") as result:
+            result.raise_for_status()
+            return True
+    except Exception:
+        return False
+
+
+def __parse_github_url(url: str) -> Optional[str]:
+    """
+    Parses GitHub url and returns a url to the root of the files.
+
+    Returns
+    -------
+    Url to the root of the files or `None` if error.
+    """
+
+    GITHUB_RAW_URL = "https://raw.githubusercontent.com"
+
+    # If does not follow format `https://github.com/account/repository`
+    if url.split(SupportedGitProviderLinks.GITHUB_URL)[1].count("/") != 2:
+        return None
+
+    return url.replace(SupportedGitProviderLinks.GITHUB_URL, GITHUB_RAW_URL)
+
+
+def __parse_gitlab_url(url: str) -> Optional[str]:
+    """
+    Parses GitLab url and returns a url to the root of the files.
+
+    Returns
+    -------
+    Url to the root of the files or `None` if error.
+    """
+
+    GITLAB_RAW_SUFFIX = "/-/raw"
+
+    # If does not follow format `https://gitlab.com/account/repository`
+    if url.split(SupportedGitProviderLinks.GITLAB_URL)[1].count("/") != 2:
+        return None
+
+    return url + GITLAB_RAW_SUFFIX
+
+
+def __parse_bitbucket_url(url: str) -> Optional[str]:
+    """
+    Parses Bitbucket url and returns a url to the root of the files.
+
+    Returns
+    -------
+    Url to the root of the files or `None` if error.
+    """
+
+    BITBUCKET_RAW_SUFFIX = "/raw"
+
+    # If does not follow format `https://bitbucket.org/account/repository`
+    if url.split(SupportedGitProviderLinks.BITBUCKET_URL)[1].count("/") != 2:
+        return None
+
+    return url + BITBUCKET_RAW_SUFFIX
+
+
+def parse_url(url: str) -> Optional[str]:
+    """
+    Tries to transform the given URL to a URL to the root of the raw git repository.
+
+    Supported Git Providers
+    -----------------------
+    `GitHub`, `GitLab` and `BitBucket`
+
+    Other Git Providers
+    -------------------
+    If the Git provider is not part of the list above, the URL is returned as-is.
+
+    Returns
+    -------
+    The transformed URL if success, otherwise `None`
+    """
+
+    parsed_url: Optional[str] = url[:-1] if url.endswith("/") else url
+    if url.startswith(SupportedGitProviderLinks.GITHUB_URL):
+        parsed_url = __parse_github_url(url)
+    elif url.startswith(SupportedGitProviderLinks.GITLAB_URL):
+        parsed_url = __parse_gitlab_url(url)
+    elif url.startswith(SupportedGitProviderLinks.BITBUCKET_URL):
+        parsed_url = __parse_bitbucket_url(url)
+
+    return parsed_url
+
+
+def __validate_attribute(
+    repo_name: str, conf: Dict[str, Optional[str]], attribute_name: str, ttype: type
+) -> bool:
```

Comment:
`valtype` seems like a more descriptive name, since you are using it to validate an attribute

Reply:
> valtype seems like a more descriptive name, since you are using it to validate an attribute

`variable_type`, remember: snake case

---

Why do we need to do this though?

Reply:
Basically it asserts that certain attributes from the toml file are the correct type
e.g: bools to be true or false, numbers to only be digits and not strings (`123` is fine but `"123"` is not), etc

Reply:
Can't we convert it during runtime though?

Reply:
sure but why bother? is `'true'` a valid boolean? what about `"True"`, `1`, `"1"`, `"yes"`?